src/RustyObjectStore.jl

```diff
@@ -719,28 +767,32 @@ Atomically writes the data bytes in `buffer` to `path`.
 function put_object(buffer::AbstractVector{UInt8}, path::String, conf::AbstractConfig)
     response = Response()
     size = length(buffer)
-    cond = Base.AsyncCondition()
-    cond_handle = cond.handle
+    ct = current_task()
+    event = Base.Event()
+    handle = pointer_from_objref(event)
     config = into_config(conf)
     while true
-        result = GC.@preserve buffer config response cond begin
+        preserve_task(ct)
+        result = GC.@preserve buffer config response event try
             result = @ccall rust_lib.put(
                 path::Cstring,
                 buffer::Ref{Cuchar},
                 size::Culonglong,
                 config::Ref{Config},
                 response::Ref{Response},
-                cond_handle::Ptr{Cvoid}
+                handle::Ptr{Cvoid}
             )::Cint
 
-            wait_or_cancel(cond, response)
+            wait_or_cancel(event, response)
 
             result
+        finally
+            unpreserve_task(ct)
         end
 
         if result == 2
             # backoff
-            sleep(1.0)
+            sleep(0.01)
             continue
         end
```

Comment:
Question: Why do we check `result == 2`, _after_ waiting? The wait_or_cancel cannot affect the value of `result`, so if result == 2 means we need to retry, why even wait?

I think this could use a comment, and/or a global constant that gives a meaningful name to whatever `2` is.